src/v3/plugins/github/relationalView.js

```diff
@@ -224,3 +219,301 @@ export class RelationalView {
+  }
+}
+
+type ReviewEntry = {|
+  +address: N.ReviewAddress,
+  +body: string,
+  +url: string,
+  +comments: N.CommentAddress[],
+  +state: Q.ReviewState,
+  +nominalAuthor: ?N.UserlikeAddress,
+|};
+
+export class Review {
+  _view: RelationalView;
+  _entry: ReviewEntry;
+  constructor(view: RelationalView, entry: ReviewEntry) {
+    this._view = view;
+    this._entry = entry;
+  }
+  address(): N.ReviewAddress {
+    return this._entry.address;
+  }
+  type(): typeof N.REVIEW_TYPE {
+    return N.REVIEW_TYPE;
+  }
+  parent(): Pull {
+    const address = this.address().pull;
+    const pull = this._view.pull(address);
+    return assertExists(pull, address);
+  }
+  body(): string {
+    return this._entry.body;
+  }
+  url(): string {
+    return this._entry.url;
+  }
+  state(): string {
+    return this._entry.state;
+  }
+  *comments(): Iterator<Comment> {
+    for (const address of this._entry.comments) {
+      const comment = this._view.comment(address);
+      yield assertExists(comment, address);
+    }
+  }
+  authors(): Iterator<Userlike> {
+    return getAuthors(this._view, this._entry);
+  }
+}
+
+type CommentEntry = {|
+  +address: N.CommentAddress,
+  +body: string,
+  +url: string,
+  +nominalAuthor: ?N.UserlikeAddress,
+|};
+
+export class Comment {
+  _view: RelationalView;
+  _entry: CommentEntry;
+  constructor(view: RelationalView, entry: CommentEntry) {
+    this._view = view;
+    this._entry = entry;
+  }
+  address(): N.CommentAddress {
+    return this._entry.address;
+  }
+  type(): typeof N.COMMENT_TYPE {
+    return N.COMMENT_TYPE;
+  }
+  parent(): Pull | Issue | Review {
+    const address = this.address().parent;
+    let parent: ?Pull | ?Issue | ?Review;
+    switch (address.type) {
+      case "PULL":
+        parent = this._view.pull(address);
+        break;
+      case "ISSUE":
+        parent = this._view.issue(address);
+        break;
+      case "REVIEW":
+        parent = this._view.review(address);
+        break;
+      default:
+        throw new Error(`Unexpected parent address: ${stringify(address)}`);
+    }
+    return assertExists(parent, address);
+  }
+  body(): string {
+    return this._entry.body;
+  }
+  url(): string {
+    return this._entry.url;
+  }
+  authors(): Iterator<Userlike> {
+    return getAuthors(this._view, this._entry);
+  }
+}
+
+type UserlikeEntry = {|
+  +address: N.UserlikeAddress,
+  +url: string,
+|};
+
+export class Userlike {
+  _view: RelationalView;
+  _entry: UserlikeEntry;
+  constructor(view: RelationalView, entry: UserlikeEntry) {
+    this._view = view;
+    this._entry = entry;
+  }
+  address(): N.UserlikeAddress {
+    return this._entry.address;
+  }
+  type(): typeof N.USERLIKE_TYPE {
+    return N.USERLIKE_TYPE;
+  }
+  login(): string {
+    return this.address().login;
+  }
+  url(): string {
+    return this._entry.url;
+  }
+}
+
+function assertExists<T>(item: ?T, address: N.StructuredAddress): T {
```

Comment:
It would be nice to be able to write

```js
function assertExists<T>(
  item: ?T,
  address: N.StructuredAddress & {+type: $Call<$PropertyType<T, "type">>}
): T {
```

but, unfortunately, [a Flow bug][bug] prevents this from working in the
case of comment parents.

[bug]: https://github.com/facebook/flow/issues/6530